eq-author-api/schema/resolvers/base.js

```diff
@@ -519,19 +519,28 @@ const Resolvers = {
       const removedOption = first(remove(answer.options, { id: input.id }));
 
       pages.forEach(page => {
-        if (!page.routing) {
+        if (!page.routing && !page.skipConditions) {
           return;
         }
 
-        page.routing.rules.forEach(rule => {
-          rule.expressionGroup.expressions.forEach(expression => {
-            if (expression.right && expression.right.optionIds) {
-              remove(
-                expression.right.optionIds,
-                value => value === removedOption.id
-              );
-            }
-          });
+        const routingExprs = page.routing
+          ? page.routing.rules.flatMap(
+              rule => rule.expressionGroup && rule.expressionGroup.expressions
+            )
+          : [];
+        const skipExprs = page.skipConditions
+          ? page.skipConditions.flatMap(
+              condition => condition && condition.expressions
+            )
+          : [];
+        const rightHandSides = [...routingExprs, ...skipExprs].map(
+          x => x && x.right
+        );
+
+        rightHandSides.forEach(right => {
+          if (right && right.optionIds) {
+            remove(right.optionIds, value => value === removedOption.id);
+          }
```

Comment:
You'll want to do this as an OR (`||`) instead of an AND (`&&`). Pages can have routing but not skip conditions, and vice versa. Having it as an AND would mean that you'll need both for this to happen. 

This also means that you'll need to pop the code for handling skip and routing in their own conditional block, too. You can probably keep your fail fast option, 'though, for pages that don't have either.

Your test likely works for skip conditions because the questionnaire JSON has routing rules on the page. It might be worth making some tests for if you have both, skip without routing and routing without skip. Your skip condition test can be repurposed into the one for if you have both.

Reply:
Thanks Matt,

This was a way of unifying the logic for the resolver - otherwise as you'd say you'd need two almost identical blocks of logic which do essentially the same thing (but with slightly different variable names). The logic as is handles both skip conditions and routing - but it obviously depends on there being routing or skip logic to process, hence the initial early `return` if there's nothing to do.

The test doesn't have routing rules & skip logic on the same page - the first page only has routing logic and the second only has skip conditions, just like in an actual questionnaire. It works in practice too in Author - handles deletion of options from both routing (as before) and skip conditions irrespective of whether other pages have skip conditions / logic enabled.

So I think it's alright. Let me know if you disagree further ;-)

Reply:
Sorry. You're absolutely right. I didn't look into it deep enough. Ignore me!

Reply:
For reference, there is a function in the utils.js that returns all the expressions for a questionnaire in a single array to  iterate over.

Reply:
Hmm, thanks Martyn. I'll have a dig. Can perhaps refactor as part of another PR.

Reply:
It's creatively named 'getExpressions'

Reply:
Very exciting. I might squeeze it into my folders update PR since that touches this file too.